[API] add User Entity

Dominskyi93 opened the pull request.

Dominskyi93 committed:
added User Entity

AleksandrLts commented on src/main/java/com/example/carsharingservice/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,33 @@
+
+    @Override
+    public User findByEmail(String email) {
+        return userRepository.findByEmail(email);
```
```suggestion
        return userRepository.findByEmail(email)orElseThrow(
        () -> new EntityNotFoundException(custom message)
        );
```

AleksandrLts commented on src/main/java/com/example/carsharingservice/service/impl/UserServiceImpl.java:
```diff
@@ -0,0 +1,33 @@
+
+    @Override
+    public User get(Long id) {
+        return userRepository.getReferenceById(id);
```
Same

AleksandrLts commented on src/main/java/com/example/carsharingservice/model/User.java:
```diff
@@ -0,0 +1,29 @@
+    @Id
+    @GeneratedValue(strategy = GenerationType.IDENTITY)
+    private Long id;
+    @Column(nullable = false)
```
add unique property

AleksandrLts commented on src/main/java/com/example/carsharingservice/controller/UserController.java:
```diff
@@ -0,0 +1,49 @@
+    @PutMapping("/{id}/role")
+    public UserResponseDto updateRole(@PathVariable Long id, @RequestParam String role) {
+        User user = userService.get(id);
+        user.setRole(Role.valueOf(role));
```
you need to check if role already exist in Role class

AleksandrLts reviewed (changes requested):
Gj, fix a little

Dominskyi93 committed:
1. replaced methods User findByEmail(String email) with Optional<User> findByEmail(String email);
2. replaced methods User findById(Long id);
 with Optional<User> findById(Long id);
3. added unique property for email
4. In the method, UserResponseDto updateRole check if the role exists

Dominskyi93 committed:
1. added changelog
2. fixed some settings

Dominskyi93 committed:
1. removed redundant dto
